### DISCUSSION TITLE
Allow users to decide whether to enable "system prompts" and "memory" for custom models

### DISCUSSION THREAD

**Discussion by CookSleep** [open-webui/open-webui#4894] [Ideas]
**Is your feature request related to a problem? Please describe.**
When I use custom models, I always expect them to fully comply with the system prompts specifically written for them to better handle a certain type of problem. However, I find that the "system prompts" and "memory" written in my account settings are also added to the custom model's system prompts, which causes them to be confused about which guidance to follow.
A very simple example:
In the "system prompts" of my account, I want the model to "output concisely and efficiently," but in a separate system prompt for a custom model used for writing novels, I set "write complex and engaging non-traditional plots." Since both the account and custom model system prompts are included in the final system prompt given to the model, the model doesn't know whether to follow "output concisely and efficiently" or "write complex and engaging non-traditional plots"!

**Describe the solution you'd like**
Add a switch to allow users to decide whether to enable "system prompts" and "memory" for custom models. It would be best to add separate switches for "system prompts" and "memory" to prevent some people from wanting "memory" to be effective for custom models.

**Describe alternatives you've considered**
Deleting the account's system prompts every time I use a custom model is still too troublesome.

**Comment by thiswillbeyourgithub** [comment]
I agree. In a more general way openwebui is really lacking granular permission control for users. I really hope it's considered among the priority feature along with bug squashing and UI enhancement. The only thing I fear for openwebui is to spread too thin and ship too many half baked features instead of a slow is smooth and smooth is fast approach.


Edit: it's on track! https://github.com/open-webui/open-webui/issues/2904
